### ISSUE TITLE
Lidl Melinera Smart LED string lights

### ISSUE THREAD

**Issue by sonar98** [dresden-elektronik/deconz-rest-plugin#3716]
## Device

- Product name: The device name as shown on the product or package.
- Manufacturer: _TZE200_s8gkrkxk
- Model identifier: TS0601
- Device type : Light

## Screenshots
<img width="195" alt="cluster-of-node" src="https://user-images.githubusercontent.com/645403/99819272-8cd34980-2b4f-11eb-9020-5f6a085c818f.png">
<img width="260" alt="node-info" src="https://user-images.githubusercontent.com/645403/99819301-96f54800-2b4f-11eb-9337-b95883eda77c.png">
<img width="396" alt="basic-cluster-1" src="https://user-images.githubusercontent.com/645403/99819218-7b8a3d00-2b4f-11eb-894d-cd87c3ba546d.png">
<img width="392" alt="basic-cluster-2" src="https://user-images.githubusercontent.com/645403/99819239-81801e00-2b4f-11eb-9512-4f4e13c7ab1d.png">

<img width="512" alt="cluster-colorcontrol-1" src="https://user-images.githubusercontent.com/645403/99819418-bf7d4200-2b4f-11eb-902b-06d005b1ea88.png">
<img width="513" alt="cluster-colorcontrol-2" src="https://user-images.githubusercontent.com/645403/99819435-c3a95f80-2b4f-11eb-971e-3496e53bb101.png">
<img width="514" alt="cluster-colorcontrol-3" src="https://user-images.githubusercontent.com/645403/99819445-c60bb980-2b4f-11eb-863e-95acc7edb3e6.png">
<img width="514" alt="cluster-colorcontrol-4" src="https://user-images.githubusercontent.com/645403/99819450-c7d57d00-2b4f-11eb-9afc-bfa6eb4a9f6c.png">
<img width="514" alt="cluster-colorcontrol-5" src="https://user-images.githubusercontent.com/645403/99819475-cc9a3100-2b4f-11eb-9d94-574fb673f116.png">

**Comment by Mimiix** [comment]
What is missing on this one?
seems to be alright?

**Comment by sonar98** [comment]
Choosing a color is not working correctly. Colors do not match the choosen color.
And the lights have buttons for different programs. These are not available in zigbee.

**Comment by Mimiix** [comment]
@Smanar Can you include this 😃?

**Comment by Smanar** [comment]
I m searching the product to have information, but a part a video on youtube, and not sure was the same.
Do you think you can act on button using zigbee ? (instead of using them manually ?)
Do you have seen something special about color ? For exemple orange instead of blue ?

For button, if you enable log, you will be probably able to see log starting with 
"Tuya debug X: xxxxxxxxxxxxxxxxxxxxx"

When using them

**Comment by sonar98** [comment]
How can I enable log? Is that possible on a RPI or windows computer?

**Comment by Mimiix** [comment]
@sonar98 From deconz gui click "Help" and click "debug view"

**Comment by sonar98** [comment]
I tried but I don't see any line starting with: **"Tuya debug X**

**Comment by Mimiix** [comment]
Even if you enable all ?

**Comment by Smanar** [comment]
You  will have this kind of log every time the device use the tuya cluster. So try to use the switches on the devices.

BTW which one color instead of full red, full blue and full green ?

**Comment by sonar98** [comment]
When I switch it on I see:
16:58:12:200 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:12:200 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:12:397 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:12:399 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:12:599 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:12:600 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:12:803 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:12:804 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:12:997 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:12:999 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:13:191 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:13:192 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:13:393 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:13:394 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:13:684 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:13:686 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:13:794 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:13:795 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:13:999 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:14:000 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:14:189 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:14:190 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:14:391 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:14:392 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:14:592 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:14:593 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:14:797 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:14:798 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:14:983 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:14:984 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:15:188 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:15:189 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:15:390 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:15:391 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:15:595 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:15:597 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:15:783 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:15:785 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:15:988 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:15:990 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:16:188 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:16:190 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:16:380 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:16:382 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:16:603 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:16:604 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:16:782 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:16:784 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1
16:58:16:984 Tuya debug 4 : Address 0x588E81FFFEDB7E9A Payload 00280101000101
16:58:16:986 Tuya debug 5 : Status: 0 Transid: 40 Dp: 257 (0x01,0x01) Fn: 0 Data 1


But when pushing the buttons nothing happens

**Comment by Smanar** [comment]
Hooow, I think we can call this spam ^^, 10 request by seconds.

And yes 0x01 0x01 is good, like you said, it s generaly the command used to turn on a device.
But useless, as the cluster 0x0006 is already working

Unfortunately, the tuya cluster is the only one possible to receive information about the switches.

Do you know if the official lidl app can change the programs too ?

**Comment by sonar98** [comment]
I don't know (yet). The Lidl app only works with the silvercrest gateway. Friday there Will be a special offer in the shop. I will try to get a silvercrest gateway to test the capabilities.

**Comment by Smanar** [comment]
Do you have a picture of the switches on the device ?

Perhaps it s memorised scene, but the "setting" of the scene can come from another device, you can change locally the programs and it s working, but nothing usefull is visible on log ? (other than tuya debug)

**Comment by sonar98** [comment]
There are two physical buttons.
1. Color
2. Function

**Comment by Smanar** [comment]
And what they are doing ? You don't see anything on log when using them (about scenes for exemple), even it s not tuya cluster ?

**Comment by alex3305** [comment]
An online [manual](https://www.lidl-service.com/cps/rde/xchg/SID-1FC8C7D5-504DC4E3/lsp/hs.xsl/product.html?id=5019460647&title=LED-verlichtingsketting&count=1) is also available for this device.

**Comment by Smanar** [comment]
Ok I hve found some informations on z2m https://github.com/Koenkk/zigbee2mqtt.io/blob/master/docs/devices/HG06467.md

But How to use those with deconz ...
Perhaps using "state/effect" for effect, "state/colorloopspeed" for speed and for color will see later ...

**Comment by sonar98** [comment]
Is this usefull?

https://github.com/Koenkk/zigbee-herdsman-converters/pull/1799

**Comment by Smanar** [comment]
Yep, but I prefer avoid to make a so hudge code for a device used only 1 month by year ^^
I have checked the deconz code, I think I will avoid to use the api command just for that, I will use an hidden command with same parameter than the z2m code, like that I will not break the deconz code, and can make all the code in a separate file.

**Comment by sonar98** [comment]
```
"5": {
		"etag": "d47e9d5b3db50b4459f4ed3e1bd203cc",
		"hascolor": true,
		"lastannounced": null,
		"lastseen": "2020-11-24T13:16Z",
		"manufacturername": "Heiman",
		"modelid": "TS0601",
		"name": "Color dimmable light 5",
		"state": {
			"alert": "none",
			"bri": 2,
			"colormode": "xy",
			"effect": "none",
			"hue": 65018,
			"on": true,
			"reachable": false,
			"sat": 127,
			"xy": [
				0,
				0
			]
		},
		"swversion": null,
		"type": "Color dimmable light",
		"uniqueid": "58:8e:81:ff:fe:db:7e:9a-01"
	}
```

**Comment by ebaauw** [comment]
The user manual is online, https://www.lidl-service.com/cps/rde/xchg/SID-1FC8C7D5-504DC4E3/lsp/hs.xsl/product.html?id=5019460647&title=LED-verlichtingsketting&count=1.

Strip is for inside and outside use.  No individual control of lights (at least from the app), but 16 pre-programmed effects.  The app seems to support selecting speed and colours for (some of the) effects. 
Looks like we'll need their gateway to sniff how to set the effects.   EUR 70 for gateway and strip, I'm tempted...

**Comment by sonar98** [comment]
Tomorrow I will receive the gateway.

**Comment by sonar98** [comment]
Some screenshots from the LIDL Home app when using the Silvercrest gateway:

![white](https://user-images.githubusercontent.com/645403/100448671-f48b1680-30b2-11eb-95f1-67a7129f7957.jpeg)
![color](https://user-images.githubusercontent.com/645403/100448680-f8b73400-30b2-11eb-96cc-21e44d8b841f.jpeg)
![scene1](https://user-images.githubusercontent.com/645403/100448688-fbb22480-30b2-11eb-9c3a-a6b9efc8180f.jpeg)
![scene2](https://user-images.githubusercontent.com/645403/100448692-fe147e80-30b2-11eb-9436-ce5dd1541887.jpeg)
![scene3](https://user-images.githubusercontent.com/645403/100448698-0076d880-30b3-11eb-9373-76b4bfa7dddb.jpeg)
![scene4](https://user-images.githubusercontent.com/645403/100448710-040a5f80-30b3-11eb-82be-59ecee8793cd.jpeg)

**Comment by ebaauw** [comment]
Got one, incl. gateway, see #3762.  Zigbee sniffer is working overtime.

The X-Mas lightstrip consists of 200 LEDs, which seem to be individually addressable, at least from the controller.  I think LEDs are chained, connecting the data in of the next LED to the data out of the previous LED, like the Neopixel or Pimoroni Blinkt! (see Homebridge RPi).  When setting the strip to a single colour, the new colour "travels through" the strip, suggesting a serial protocol.  Same for the occasional erroneous setting of a single LED.

I'm not sure of this detailed level of control is exposed over Zigbee, though.

The strip seems to be controlled from cluster 0xEF00, but no _Manufacturer Specific_ bit.  Any change (be it white, colour or scene) results in a 0x00 command, followed by one or more 0x01 commands.  The commands have a variable payload, up to 56 bytes.  Even combined, the payloads doesn't seem to be large enough to hold the RGB and brightness values per LED.  Presumably, the controller uses the payload to generate the serial signal.

**Comment by Smanar** [comment]
@ebaauw, the protocole is already sniffed and decrypted (look at z2m project) it use classic tuya command.

My problem is include this device without adding 5/6 command in deconz core for only one device, used 1 month by year.

I think I will use only 1 command, that take aall params, and put the code in tuya file.

**Comment by Smanar** [comment]
The command to use are

//command
#define MELINERA_MODE 2
#define MELINERA_EFFECT 6

//Mode
#define MELINERA_MODE_EFFECT 0x02


//Effect
QByteArray effect = QByteArray("\x48\x50", 2); // "02" for rainbow
QByteArray speed = QByteArray("\x49\x48", 2); // speed = 10
QByteArray color= QByteArray("\x52\x52\x52", 2); // color RGB 44,44,44

data = effect  + speed + color

SendTuyaRequest( taskRef, taskType , DP_TYPE_ENUM, MELINERA_MODE  , MELINERA_MODE_EFFECT)
SendTuyaRequest( taskRef, taskType , DP_TYPE_STRING, MELINERA_EFFECT , data)

**Comment by sonar98** [comment]
if you tell me how, I can test.

**Comment by ebaauw** [comment]
OK with help from @Smanar , pointing me to earlier work by @Koenkk, I think I now understand how the X-mas lightstrip works.  

The device seems to be "write-only": you can set it's state, but it doesn't report its current state.  The app doesn't reflect the actual state, when changing the state manually, using the buttons on the controller.  As advertised, the device supports 16 effects, which, depending on the effect, can be modified for speed and up to 6 colours used.  The strip can also be set to a single colour for all LEDs.  While the LEDs are individually addressable by the controller, this level of control doesn't seem to be exposed over Zigbee (for sure the LIDL Home app isn't using this).

The strip uses Tuya cluster 0xEF00.  The gateway sends a 0x00 command to set the lightstrip state; the lightstrip returns a 0x01 command, echoing the state.  Basically, the payload of these commands follows a TVL16-like scheme.    It consists of a one-byte status (always? 0x00), a one-byte sequence number, and one or more TVL16 encoded "attributes".  The "type" consists of one-byte attribute ID (data point in Tuya speak), followed by a one-byte data type.  The length is two-byte, big-endian.  The encoding of the value obviously depends on the data type.

The lightstrip supports three modes: white, colour, and effect.  It seems to keep separate states for each mode.  So does the LIDL Home app, but as the lightstrip is "write-only" the states aren't synced when updating the state manually.

The lightstrip supports the following attributes:

attribute  | data type | length | description
--- |  --- | ---- | ---
0x01 | 0x01 (bool) | 1 | On/Off<br>Values: 0x00: off, 0x01: on
0x02 | 0x04 (enum) | 1 | Mode<br>Values: 0x00: white, 0x01: colour, 0x02: effect
0x03 | 0x02 (number) | 4 | Brightness (in 0.1%).<br>Value is a big-endian `u32`.
0x05 | 0x03 (string) | 4+4+4 | Hue (in °), sat (in 0.1%), bri (in 0.1%).<br>Values are ascii-encoded, big-endian hex values.
0x06 | 0x03 (string) | 2+2+_n_*6| Effect (0x00 - 0x0F), speed (in %, or 0 if n/a), _n_ Colours in RGB (0-255).<br>Values are ascii-encoded, big-endian hex values.

Attribute 0x03 seems applicable only in white mode; 0x05 only in colour mode; and 0x06 only in effect mode.  The gateway typically sends attributes 0x01, 0x02, and 0x06 in a single 0x00 command.  Note that in colour mode, 16-bit hue/sat values are used, whereas in effect mode, 8-bit RGB values are used.  You cannot control brightness while in effect mode.

I haven't yet tried the strip with deCONZ, but I seriously doubt that the _Color Control_ cluster would be usable.  I don't think the lightstrip supports the CIE 1931 colour space, so the `xy` values would be bogus.

Supporting the lightstrip in the REST API will be challenging from a design perspecitive and ugly from an implementation perspective.
- I don't think we currently expose `mode` for lights.  We could use `"effect": "none"` to switch to colour mode, and `{"effect": "`_name_`"}` to switch to effect mode.  I'm not sure how to switch to white mode.
- It's probably easiest to use `hue` and `sat` in colour mode, but this will be hell for API clients, who (should) use `xy`.  I suppose we better use `xy` values, and transform them to RGB `hue`/`sat` values.  On the other hand, setting `sat` to `0` would be a natural way for switching to white mode.
- Exposing the effects using `effect` with new values for the effect names seems doable.  We already have `colorloopspeed`; best rename that to `effectspeed` (`speed` is already taken by fans, unfortunately).  The effect colours are best modelled as an `effectcolors` array of RGB values.  I suppose few API clients will go to this level of detail (I, for one, wouldn't know how to expose this to HomeKit), so accepting `xy` values and translating them to RGB seems to have little benefit.

**Comment by ebaauw** [comment]
Some more observations:
- The LIDL Home app correctly shows the lightstrip as on, after power cycling it.  I guess it uses the _Device Announcement_ for this;
- Paired the lightstrip with deCONZ.  The _On/Off_ and _Level Control_ clusters seem to be doing the right thing.
  - Both _OnOff_ and _Current Level_ send attribute reports;
  - When setting the level, the lightstrip exits effect mode;
  - _On_, _Off_, _Toggle_, _On with Timed Off_ seem to work just fine;
  - _Move to Level (with On/Off)_ turns the lightstrip on and off, as it should;
  - _Step_ and _Step (with On/Off)_ seem to work fine as well;
  - However, _Current Level_ isn't updated on _Move_ and _Move (with On/Off)_.  On _Stop_, _Current Level_ is set to 100.  These are the commands used by the Trådfri on/off switch.
  - Adding the device to a controller group works as expected.  Note that the device state isn't updated correctly when the controller uses _Move_ / _Stop_, see previous point.
- _Color Control_ actually seems almost usable.  _Move to Hue and Saturation_ as well as _Move to Color_ seem to work, but the colours are too white, not saturated enough.  Seems like the lightstrip is in a sort of combined white/colour mode.  Setting level to 1 shows better colours (but obviously not bright enough).  The lightstrip doesn't support _Move to Color Temperature_.
  - The device seems to ignore transition time throughout.

There's something weird with the API, maybe another Tuya hack, @Smanar?  It doesn't seem to accept updates to `bri`:
```
$ ph put -v /lights/13/state '{"bri": 127}'
ph put: warning: api error 6: parameter not available
[
  {
    "error": {
      "address": "/lights/13/state/on",
      "description": "parameter not available",
      "type": 6
    }
  }
]
```
Updating `bri` through a group works as expected.

**Comment by ebaauw** [comment]
There seems to be something in `general.xml` to construct a 0xEF00 command from _Status_, _TransID_, _Dp_, _fn_, and _data_.  However, there's no chance in Hell that that'll ever work: the payload isn't ZCL.  As I mentioned above, _Dp_ is actually two fields, the data point and data type.  I think _fn_ is the high byte of the length; the first byte in _data_ is actually the low byte of length.

**Comment by ebaauw** [comment]
Looks like I can control the lightstrip through the CLI plugin.  It responds with several dozen of 0x01 commands, since deCONZ doesn't send a default response.
```
$ echo "zclcmd 0xfa81 1 0xEF00 0000000101000101" | nc localhost 5008
<-APS attr 0x677B 1 0x0000 0x0006 0x42 
zclcmd 0xfa81 1 0xEF00 0000000101000101
 --> send OK
<-ZCL serverToClient 0xFA81 1 for cluster 0xEF00 00 00 01 01 00 01 01 
<-ZCL serverToClient 0xFA81 1 for cluster 0xEF00 00 00 01 01 00 01 01 
```
Note that the last parameter is the Zigbee command followed by the payload:
```
00 00 00 01 01 0001 01
|  |  |  |  |  |    + Value 0x01: true
|  |  |  |  |  + Length: 1
|  |  |  |  + Data type: 0x01: boolean
|  |  |  + Data point 0x01: on/off
|  |  + Transaction ID: 0x00
|  + Status: 0x00
+ Zigbee command 0x00: set
```

Similarly, setting the twinkle effect at a speed of 50:
```
$ echo "zclcmd 0xfa81 1 0xEF00 0000000603000430343332" | nc localhost 5008
zclcmd 0xfa81 1 0xEF00 0000000603000430343332
 --> send OK
<-APS attr 0x677B 1 0x0000 0x0006 0x42 
<-ZCL serverToClient 0xFA81 1 for cluster 0xEF00 00 00 06 03 00 04 30 34 33 32 
<-ZCL serverToClient 0xFA81 1 for cluster 0xEF00 00 00 06 03 00 04 30 34 33 32 
```
Decoding of the parameter:
```
00 00 00 06 03 0004 3034 3332
|  |  |  |  |  |    + value: 0x30403332 = ascii "0432" = 0x04 (twinkle), 0x32 (speed: 50)
|  |  |  |  |  + Length: 4
|  |  |  |  + Data type: 0x03: string
|  |  |  + Data point 0x06: effect
|  |  + Transaction ID: 0x00
|  + Status: 0x00
+ Zigbee command 0x00: set
```

**Comment by Smanar** [comment]
>There's something weird with the API, maybe another Tuya hack, @Smanar? It doesn't seem to accept updates to bri:

It's not realy because a hack ^^, ATM this device is reconized as switch (cf the other issue), but Tuya switch, so this part need to be skipped by your device

```
    else if (UseTuyaCluster(taskRef.lightNode->manufacturer()))
    {
        //window covering
        if ((taskRef.lightNode->manufacturer() == QLatin1String("_TYST11_wmcdj3aq")) ||
            (taskRef.lightNode->manufacturer() == QLatin1String("_TZE200_xuzcvlku")) ||
            (taskRef.lightNode->manufacturer() == QLatin1String("_TZE200_wmcdj3aq")) ||
            (taskRef.lightNode->manufacturer() == QLatin1String("_TYST11_xu1rkty3")) )
        {
            return setWindowCoveringState(req, rsp, taskRef, map);
        }
        //switch
        else
        {
            return setTuyaDeviceState(req, rsp, taskRef, map);
        }
    }
```

Your device use tuya cluster, but is not covering, so the code use the switch part, just need to add
```
elseif ((taskRef.lightNode->manufacturer() == QLatin1String("YOURMANUFACTURENAME")) 
{
}
```

And this part will be skipped for this device

And for " implementation perspective", I have checked the code, I think using using "ffect" and "colorspeed" is a bad idea, because :
- Need lot of modification in code just for 1 device used 1 month by year
- Can provoque problem in the core, thoses field can be used elsewhere.

In my mind was easier to make an hidden command like
`PUT /api/<apikey>/lights/ID with {"effect":["rainbow",20,12345,6578]}`
For exemple

Will be harder for integration in third app, but even if we use standard deconz param it will be, so ....

With this way, less code to do, less dangerous for core, and can put all the code in separate file (the tuya one)

**Comment by sonar98** [comment]
@Smanar about third party... It would be nice if the device could be used in domoticz.

**Comment by Smanar** [comment]
Lol, will see, christmas is just in 28 days.

**Comment by GH2user** [comment]
Maybe the HG06467 protocol is useful to you guys, [https://pastebin.com/pcgWVjCp](url) as it describes all options for the Xmas lights.

**Comment by Goldwing1973** [comment]
> Maybe the HG06467 protocol is useful to you guys, [https://pastebin.com/pcgWVjCp](url) as it describes all options for the Xmas lights.

Your URL doesn’t work, this link should work.
[https://pastebin.com/pcgWVjCp](https://pastebin.com/pcgWVjCp)

**Comment by MichielMag** [comment]
I wanna chime in real quick on that pastebin: that was my first look at the protocol a while ago. When I made it I didn't really understand the values. Finally it clicked that the tuya protocol defined datatypes and that one of them (0x03) is string. Which means that 0x30 = '0'. Therefor a scene code is for example (0x3031 = '01'). Same goes for the speed and the HSB string in the color command.

What I'm currently missing (which is also missing from the implementation I did for Z2M) are :
- color temperature (the packaging claims the LED string has support for it)
- RGB 'color' (when you press the physical 'C' button enough times, it changes the string to a classic colored christmas LED string)

I've emailed Lidl support with the question if these things will be added to the app later. But I also have my hoped on you guys to maybe figure it out, so I'm subscribing to this topic :)

**Comment by ebaauw** [comment]
> color temperature (the packaging claims the LED string has support for it)

No proof of that sniffing the LIDL gateway.  The LEDs appear to be RGBW, so no physical support for colour temperature either.

> RGB 'color' (when you press the physical 'C' button enough times, it changes the string to a classic colored christmas LED string)

As I mention above, the light strip's 0xEF00 cluster is pretty much "write-only".  You can set its state, but it doesn't report its current state.  The 0x01 command in reply to the 0x00 command simply echoes the 0x00 payload.  When setting the state through the _On/Off_, _Level Control_, or _Color Control_ clusters, those attributes are updated (except for _Move_, see above); but those attributes aren't updated when setting the 0xEF00 cluster.

**Comment by ebaauw** [comment]
Above PR adds REST API support.

```
{
  "etag": "765141a6915c0c78b8399aea0f3096ee",
  "hascolor": true,
  "lastannounced": null,
  "lastseen": "2020-11-29T22:13Z",
  "manufacturername": "_TZE200_s8gkrkxk",
  "modelid": "TS0601",
  "name": "Color dimmable light 13",
  "state": {
    "bri": 2,
    "colormode": "hs",
    "effect": "tinkle",
    "hue": 0,
    "on": true,
    "reachable": true,
    "sat": 0
  },
  "swversion": null,
  "type": "Color dimmable light",
  "uniqueid": "58:8e:81:ff:fe:d8:83:cc-01"
}
```

To control the Xmas light strip:
- Set `on` for on/off.  I added support for `ontime`, but the strip doesn't seem to support it;
- Set `hue` (0 - 65535), `sat` (0 - 254), and `bri` (0 - 254) as HSL values for colour mode;
- Set `sat` to 0 and `bri` (0-254)  for white mode;
- Set `effect` and, optionally, `effectSpeed` and `effectColours` for effect mode, e.g.
  `{"effect": "flag", "effectSpeed": 10, "effectColours": [[255,0,0],[0,255,0],[0,0,255]]}`
  The effects are: `"steady"`, `"snow"`, `"rainbow"`, `"snake"`, `"tinkle"`, `"fireworks"`, `"flag"`, `"waves"`, `"updown"`, `"vintage"`, `"fading"`, `"collide"`, `"strobe"`, `"sparkles"`, `"carnival"`, `"glow"`;
- Set `"effect": "none"` to revert to previous mode (white or colour).

**Comment by canim60** [comment]
> 
> 
> Above PR adds REST API support.
> 
> ```
> {
>   "etag": "765141a6915c0c78b8399aea0f3096ee",
>   "hascolor": true,
>   "lastannounced": null,
>   "lastseen": "2020-11-29T22:13Z",
>   "manufacturername": "_TZE200_s8gkrkxk",
>   "modelid": "TS0601",
>   "name": "Color dimmable light 13",
>   "state": {
>     "bri": 2,
>     "colormode": "hs",
>     "effect": "tinkle",
>     "hue": 0,
>     "on": true,
>     "reachable": true,
>     "sat": 0
>   },
>   "swversion": null,
>   "type": "Color dimmable light",
>   "uniqueid": "58:8e:81:ff:fe:d8:83:cc-01"
> }
> ```
> 
> To control the Xmas light strip:
> 
>     * Set `on` for on/off.  I added support for `ontime`, but the strip doesn't seem to support it;
> 
>     * Set `hue` (0 - 65535), `sat` (0 - 254), and `bri` (0 - 254) as HSL values for colour mode;
> 
>     * Set `sat` to 0 and `bri` (0-254)  for white mode;
> 
>     * Set `effect` and, optionally, `effectSpeed` and `effectColours` for effect mode, e.g.
>       `{"effect": "flag", "effectSpeed": 10, "effectColours": [[255,0,0],[0,255,0],[0,0,255]]}`
>       The effects are: `"steady"`, `"snow"`, `"rainbow"`, `"snake"`, `"tinkle"`, `"fireworks"`, `"flag"`, `"waves"`, `"updown"`, `"vintage"`, `"fading"`, `"collide"`, `"strobe"`, `"sparkles"`, `"carnival"`, `"glow"`;
> 
>     * Set `"effect": "none"` to revert to previous mode (white or colour).

Is that now Working? how can i set effects on Deconz? like Snow Rainbow?

**Comment by Smanar** [comment]
@MichielMag I have discovered the tuya protocol with this link https://medium.com/@dzegarra/zigbee2mqtt-how-to-add-support-for-a-new-tuya-based-device-part-2-5492707e882d

There is some improvement now on the DP part (this one is composed by 2 part in reality, 1 normalized and 1 device dependant), but most of the protocol is explained here

**Comment by FunkOdyssey2001** [comment]
Does anyone have any suggestion on how to mix a warm light just using saturation?

**Comment by Smanar** [comment]
@sonar98 if you can pm me on discord or domoticz forum ( I m  Thorgal789) for it s integration on domoticz pls.

**Comment by snozzlebert** [comment]
Is it possible to have a static "flag" or "rainbow" effect?

**Comment by ebaauw** [comment]
> Does anyone have any suggestion on how to mix a warm light just using saturation?

I’d try something yellow-ish with a lower `sat` value.  Should be a `hue` value around 60°, or API value 60 * 65535 / 360.

> And could it be that scenes aren't working?

I assume you mean Zigbee scenes.  Haven’t tried these yet, but I doubt they’ll work, except maybe for on/off.  I don’t expect much use for Zigbee groups either.

> Is it possible to have a static "flag" or "rainbow" effect?

Can you set these from the LIDL app, when the strip is connected to their gateway?

**Comment by snozzlebert** [comment]
> > Is it possible to have a static "flag" or "rainbow" effect?
> 
> Can you set these from the LIDL app, when the strip is connected to their gateway?

I don't own a Lidl gateway and don't have the Lidl app. I bought the light string yesterday and I'm just curious if someone else knows it. Thanks for building the API integration in deCONZ!

**Comment by sonar98** [comment]
I'm testing if it is working now but I get aan error.
```
`{
  "error": {
    "address": "/lights/5/state",
    "description": "invalid value, snake, for parameter, effect",
    "type": 7
  }`
```

**Comment by ebaauw** [comment]
Are you sure you're using the right light resource?  If so, please list the resource.

**Comment by sonar98** [comment]
```
`{
  "etag": "f5f8d2a1f2919d44ddde4b290339d5c8",
  "hascolor": true,
  "lastannounced": "2020-12-01T11:36:47Z",
  "lastseen": "2020-12-01T11:48Z",
  "manufacturername": "Heiman",
  "modelid": "TS0601",
  "name": "Color dimmable light 5",
  "state": {
    "alert": "none",
    "bri": 180,
    "colormode": "hs",
    "effect": "none",
    "hue": 43862,
    "on": true,
    "reachable": true,
    "sat": 100,
    "xy": [0.6994, 0.2959]
  },
  "swversion": null,
  "type": "Color dimmable light",
  "uniqueid": "xxxxxxxxx"
}`
```

**Comment by ebaauw** [comment]
The `manufacturername` incorrectly lists as `"Heiman"`.  Consequenlty, the Xmas light strip isn’t recognised nor exposed correctly.  Did you delete the resource and re-pair the lightstrip after updating deCONZ?

**Comment by sonar98** [comment]
yes I readded the device and it is still recognized as Heiman.

![image](https://user-images.githubusercontent.com/645403/100742791-a3916000-33db-11eb-9ad8-2cf2b1f63a98.png)

**Comment by mawa85** [comment]
Hi
I also have an issue, Paired with the older DEconz release, now after the update I deleted the devices, put the led strip into pairing mode but it is not found by deconz anymore, the old detected devices reappear, but the strip does not leave pairing mode

**Comment by ebaauw** [comment]
Did you delete the resource before re-pairing the device?  Did it get a new resource ID?  I double-checked on v02.07.00: the device is recognised correctly.

**Comment by Smanar** [comment]
I realy hate this feature that set manufacture name according to manufacture number ...

**Comment by mawa85** [comment]
sorry, I updated deconz but forgot to flash the new firmware, my bad. working now. thanks

**Comment by ebaauw** [comment]
Above commit provides some minor enhancements and bug fixes:
- Remove support for `ontime`, as the Xmas lightstrip doesn't support it;
- Add support for `bri_inc`;
- Keep _Current Level_ attribute in _Level Control_ cluster up to date, so direct changes by controllers are reflected in the API;
- Bug fix: Mueller light effect would not stick in API state;
- Bug fix: effect is called `twinkle` (not: `tinkle`).

While it's possible to control the Xmas light strip directly by adding it to a controller group, this does cause some weird behaviour.  When in effect mode, the light strip will revert to white or colour mode, but this is not reflected in the API (`effect` isn't reset to `"none"`).  Also, any colour setting is not reflected in the API.  Brightness and On/Off changes should be reflected correctly in the API.

It might be preferable to use gateway rules instead of direct control for the Xmas lightstrip, hence the added support for `bri_inc`.  Any update through the API should be reflected correctly.

**Comment by Staubgeborener** [comment]
Wow, that's cool @ebaauw, great work. Now i'm thinking about buying one of those because ... it works.

**Comment by canim60** [comment]
i will buy one tommow, i will check it up

**Comment by Somberland** [comment]
thanks for your great work @ebaauw! bought one like 20 minutes ago due to the changelog on homebridge-hue :D will test after work

**Comment by canim60** [comment]
@ebaauw how can i change the effects? I have only these objects on iobroker 

![grafik](https://user-images.githubusercontent.com/66805899/101070532-c8343600-359b-11eb-9ebb-fee97883930c.png)
![grafik](https://user-images.githubusercontent.com/66805899/101070585-d5512500-359b-11eb-8247-99899e94a2cb.png)

**Comment by ebaauw** [comment]
Thru the API.  I have no clue about iobroker.

**Comment by chris8519** [comment]
einfach dort den effekt eigeben .... das klappt

**Comment by canim60** [comment]
> 
> 
> einfach dort den effekt eigeben .... das klappt

tut aber leider nichts, habe snow oder etliche andere die oben auf liste stehen eingegeben, ohne funktion

![grafik](https://user-images.githubusercontent.com/66805899/101072700-b43e0380-359e-11eb-9fa5-0d859bbdde80.png)

**Comment by canim60** [comment]
> 
> 
> Thru the API. I have no clue about iobroker.

can you tell me how i do it ? with Node?

**Comment by gentoair** [comment]
Hallo can you explain me please, where i can insert this code? If i right understand this is for Conbee. 
Where i can find last version of this mode? 
Thank you verz much ;) @ebaauw

**Comment by ebaauw** [comment]
My first PR has been released in v2.7.0-beta.  For my second PR you need to wait or compile the REST API plugin yourself, see README.

For help on the API, see https://dresden-elektronik.github.io/deconz-rest-doc/.

**Comment by Krolykke** [comment]
Wow @ebaauw your work and efforts impress me time after time, THANK YOU! I am currently @ 2.05.88 / 15.10.2020 and not very keen on implementing beta software. Do you have an idea when this will be released as non beta and come as an update in HA?

**Comment by dasoho** [comment]
Thank you very much @ebaauw for spending lots of effort for getting this thing supported. It worked just out of the box :-)
There is just one "mode" which is selectable on the controller but which I cannot reach by the API:
If I select the colour mode on the controller (by "F" button) and then using the colour button ("C") to switch through the colours I get: red, green, blue, white, yellow, purple, cyan, coldwhite, and **multicolour**. In **multicolour** it seems that there are 6 colors defined which are randomly spread over the LED chain.
Is there a possibility to reach this "steady" **multicolour** mode via the API?
Actually, this mode is the one I would like most as it is the most classical christmas tree lighting :-)

**Comment by Hedda** [comment]
@ebaauw you do not happen to use [ZHA in Home Assistant](https://www.home-assistant.io/integrations/zha/) (depends on [zigpy](https://github.com/zigpy/zigpy) and [zha-device-handlers](https://github.com/zigpy/zha-device-handlers)) as well as deCONZ? 

Anyway, for reference, there is a similar discussion about these Xmas-lights for ZHA here -> https://github.com/zigpy/zha-device-handlers/issues/604

**Comment by Hedda** [comment]
Does anyone have much better pictures of the Christmas lights themselves and all their control/power-adapters to help a search Chinese online stores in China for a product that use the same internals so people in any country can order directly from China?

https://www.lidl.de/de/melinera-lichterkette-zigbee-smart-home/p360021

https://www.lidl.nl/nl/p/smart-home/200-rgb-led-verlichtingsketting/p43246

It sounds like most if not all of Lidl's Zigbee products are rebranded OEM or while-label version devices that are actually manufactured by Tuya or others in China, so thinking that for people not living in Europe it might be easiest to find the same product sold under a different brand from a Chinese online store and order it to the USA directly from China?

Just a few other companies who are known for rebranding and selling at least a couple OEM or while-label Zigbee devices (and WiFi devices) from example Tuya and HEIMAN are Zemismart, Alecto, Bandi, Binthen, Blitzwolf, Malmbergs, Gosund, Eachen, Moes, Sonoff, Teckin, Earda, LEELKI, Larkkey, Lonsonho, LoraTa, Norklmes, SHOJZJ, Samotech, Smart9, Somgoms, TUYATEC, UseeLink, Vrey, Woox, Yushun, and CR Smart Home,

Many such rebranded OEM or while-label Zigbee devices manufactured by Tuya can be found in zigbee2mqtt and deconz lists:

https://www.zigbee2mqtt.io/information/supported_devices.html

https://github.com/dresden-elektronik/deconz-rest-plugin/wiki/Supported-Devices

**Comment by Somberland** [comment]
@Hedda hope those helps :) from the lights I bought yesterday in Lidl Germany. 
If you need more just ping me :D 

![IMG_6068](https://user-images.githubusercontent.com/5808345/101150808-0de71200-3621-11eb-9773-fdd88829e6ba.jpeg)
![IMG_6067](https://user-images.githubusercontent.com/5808345/101150825-13445c80-3621-11eb-9d92-552ef9a0c95d.jpeg)
![IMG_6066](https://user-images.githubusercontent.com/5808345/101150828-14758980-3621-11eb-8bca-e14eef8d6955.jpeg)
![IMG_6065](https://user-images.githubusercontent.com/5808345/101150830-15a6b680-3621-11eb-9344-30f6daa0b13f.jpeg)
![IMG_6064](https://user-images.githubusercontent.com/5808345/101150833-163f4d00-3621-11eb-8870-aa11ad8b58df.jpeg)
![IMG_6063](https://user-images.githubusercontent.com/5808345/101150838-16d7e380-3621-11eb-89d2-f58d29305142.jpeg)
![IMG_6062](https://user-images.githubusercontent.com/5808345/101150842-17707a00-3621-11eb-9e4e-baaabe4beced.jpeg)
![IMG_6060](https://user-images.githubusercontent.com/5808345/101150843-18091080-3621-11eb-84a1-0e28a73d48c9.jpeg)
![IMG_6059](https://user-images.githubusercontent.com/5808345/101150847-18a1a700-3621-11eb-9e83-d57a5284bfd2.jpeg)
![IMG_6061](https://user-images.githubusercontent.com/5808345/101150854-1a6b6a80-3621-11eb-8689-ecb3f21b2793.jpeg)
![IMG_6058](https://user-images.githubusercontent.com/5808345/101150855-1b9c9780-3621-11eb-95e4-e78f19f90b43.jpeg)

**Comment by Hedda** [comment]
Slightly off-topic, love the Twinkly app concept of using the phone's camera to AR 3D mapping similar Christmas light strings:

https://www.twinkly.com/knowledge/map-your-lights-2/

Twinkly's app uses Augmented-Reality together with your phone's camera to first 3D map each and every lightbulb in the Christmas light string and then let you "paint" each and every one of those lightbulbs individually with their app on your phone.

https://www.youtube.com/watch?v=xMJ3bLBtGaQ&ab_channel=Twinkly-SmartDecoration

**Comment by Krolykke** [comment]
I tried to open mine but gave up. It's two boxes, the outer one can be forced apart, the inner one seems filled with hot glue and seems impossible to open without destroying anything.
> 
> 
> 
> 
> @Hedda hope those helps :) from the lights I bought yesterday in Lidl Germany. 
> 
> If you need more just ping me :D 
> 
> 
> 
> ![IMG_6069](https://user-images.githubusercontent.com/5808345/101151194-9796df80-3621-11eb-8582-2bc9bbc9f848.jpeg)
> 
> ![IMG_6068](https://user-images.githubusercontent.com/5808345/101150808-0de71200-3621-11eb-9773-fdd88829e6ba.jpeg)
> 
> ![IMG_6067](https://user-images.githubusercontent.com/5808345/101150825-13445c80-3621-11eb-9d92-552ef9a0c95d.jpeg)
> 
> ![IMG_6066](https://user-images.githubusercontent.com/5808345/101150828-14758980-3621-11eb-8bca-e14eef8d6955.jpeg)
> 
> ![IMG_6065](https://user-images.githubusercontent.com/5808345/101150830-15a6b680-3621-11eb-9344-30f6daa0b13f.jpeg)
> 
> ![IMG_6064](https://user-images.githubusercontent.com/5808345/101150833-163f4d00-3621-11eb-8870-aa11ad8b58df.jpeg)
> 
> ![IMG_6063](https://user-images.githubusercontent.com/5808345/101150838-16d7e380-3621-11eb-89d2-f58d29305142.jpeg)
> 
> ![IMG_6062](https://user-images.githubusercontent.com/5808345/101150842-17707a00-3621-11eb-9e4e-baaabe4beced.jpeg)
> 
> ![IMG_6060](https://user-images.githubusercontent.com/5808345/101150843-18091080-3621-11eb-84a1-0e28a73d48c9.jpeg)
> 
> ![IMG_6059](https://user-images.githubusercontent.com/5808345/101150847-18a1a700-3621-11eb-9e83-d57a5284bfd2.jpeg)
> 
> ![IMG_6061](https://user-images.githubusercontent.com/5808345/101150854-1a6b6a80-3621-11eb-8689-ecb3f21b2793.jpeg)
> 
> ![IMG_6058](https://user-images.githubusercontent.com/5808345/101150855-1b9c9780-3621-11eb-95e4-e78f19f90b43.jpeg)
> 
> 

![image](https://user-images.githubusercontent.com/31382760/101151675-4affd400-3622-11eb-9592-2e22d96bc0c6.jpeg)

**Comment by Hedda** [comment]
> @Hedda hope those helps :) from the lights I bought yesterday in Lidl Germany.
> If you need more just ping me :D

Any chance that you would be willing to also open up the control-box and take pictures of the circuit board inside it? Chances are that pictures of the circuit board (both sides if possible) could lead to us faster finding matching OEM/white-label product(s).

**Comment by eric1905** [comment]
> My first PR has been released in v2.7.0-beta. For my second PR you need to wait or compile the REST API plugin yourself, see README.
> 
> For help on the API, see https://dresden-elektronik.github.io/deconz-rest-doc/.

So I am using deconz with Phoscon and conbee. How can I update everything that it is working? In Phoscon it says: 
Version: 2.05.84 / 14.9.2020
Firmware. 26660700

**Comment by Krolykke** [comment]
> So I am using deconz with Phoscon and conbee. How can I update everything that it is working? In Phoscon it says:
> Version: 2.05.84 / 14.9.2020
> Firmware. 26660700

Are you yousing deCONZ as Homeassistant integration?

**Comment by eric1905** [comment]
I use it as a OpenHAB integration

**Comment by ebaauw** [comment]
> Is there a possibility to reach this "steady" multicolour mode via the API?

Can you reach it through the LIDL app?  As far as I can tell, the _Steady_ scene only takes a single colour.  The strip does not report its actual state, so our only change would be catching the LIDL app and gateway in the act of setting this mode.

The LIDL gateway does some magic when pairing the device (writing some manufacturer-specific attributes, sending some Tuya commands with a very small payload).  The LED strip responds with some Tuya commands as well.  However this doesn't seem to influence the working of the strip.  I tried sending all Tuya commands from 0x00 to 0x1F (using the CLI plugin), to see if one of them could lure the strip into reporting its state, but no joy.

> you do not happen to use ZHA in Home Assistant (depends on zigpy and zha-device-handlers) as well as deCONZ?

No, I only use HomeKit.  I don't even use Phoscon.

> Any chance that you would be willing to also open up the control-box and take pictures of the circuit board inside it? 

The circuit board would likely contain a Zigbee chip and a microcontroller (or a single chip combining both functions).  The microcontroller generates the serial signal to the LEDs, controlling each LED individually.  However, it only accepts a fixed number of effects over Zigbee, so we cannot control each LED individually over Zigbee.  The serial signal is write-only by nature.  The microcontroller is write-only by (poor) design, maybe limited by the resources on the board.  So there's no way to get the current state of the microcontroller.

**Comment by Smanar** [comment]
Someone can tell me if RGB color are working fine using phoscon ?
Using "sat" slider.

**Comment by Somberland** [comment]
BTW if using those Lights with HomeKit (homebridge): do I need iOS 14 for them? ![image](https://user-images.githubusercontent.com/5808345/101247889-36076b80-371c-11eb-9e10-55890c075b49.png)

**Comment by grrrmml** [comment]
First of all thanks to @ebaauw and everybody participating here to add support for the new 'xmas' device!

There ist one issue with the device identification: according to the initial data the manufacturername currently has to be "_TZE200_s8gkrkxk". 

I have seen this name only once, when i included the device for the FIRST time, on my Hue bridge (luckily I did not yet delete the device data). 
All other recent inclusions on the Conbee II ended up with manufacturername "Heiman", like also seen in the json data from @sonar98 (from 9 days ago). On the screenshot from @Somberland it is shown (translated? raw data?) as "Melinera".

I can confirm the modelid is "TS0601". But I have no knowledge if modelid TS0601 alone would be sufficient to identify the device (and whether using the effect routine on other devices could harm them).

After building the code locally with an additional OR condition with Heiman and TS0601 within isXmasLightStrip, i can use the effects via API. (Before that I had already updated Conbee II firmware to version 26680700, no idea if that was also a requirement for it to work.)

Maybe @ebaauw  could add a fix to his code. 

Thanks again for all the efforts - if I can aid with some tests, I'd be glad to participate, as I am used to squeeze all out of the (Hue) API - deconz is still very knew to me. And because I haven't written any C code for the last 5 years, that would not yet be a point to contribute.

@Smanar: "sat" basically works, but the colors (I just tested red) were only saturated when "bri" was between 16 and 39. Increasing "bri" from 39 to 40 then lead to a green tone with very low brightness. The color changes do really behave a bit odd. 
I will probably do some more (scripted) tests using the API, to see what is really going on and what is "good" to do and what is "bad". 

With best regards,
grrrmml

**Comment by Smanar** [comment]
"Heiman" again ?
I have removed it for VENDOR_EMBER how it can be possible again ?

**Comment by ebaauw** [comment]
> BTW if using those Lights with HomeKit (homebridge): do I need iOS 14 for them? Get this "not supported" error

Not a clue where that comes from.  Could you attach Homebridge Hue’s debug dump file?

> On the screenshot from @Somberland it is shown (translated? raw data?) as "Melinera".

That’s courtesy of Homebridge Hue.

> I can confirm the modelid is "TS0601". But I have no knowledge if modelid TS0601 alone would be sufficient to identify the device (and whether using the effect routine on other devices could harm them).

No, other devices use that model id as well.  We need the manufacturer name (which obviously isn’t) to identify the device.

> "sat" basically works, but the colors (I just tested red) were only saturated when "bri" was between 16 and 39. Increasing "bri" from 39 to 40 then lead to a green tone with very low brightness. The color changes do really behave a bit odd.

Then you’re not running my code, or the device isn’t recognised so my code doesn’t kick in.  Be sure to re-pair the light strip after upgrading to v2.7.0.

**Comment by grrrmml** [comment]
Hello again, and thank you for your quick and helpful replies.
@Smanar I haven't worked with the sql db of deconz yet (is there some documentation for it?) so I just reset the gateway.
Now I got "_TZE200_s8gkrkxk" on the first and second inclusion, and model TS0601 resolved to "Thermostat".
@ebaauw The mentioned effect with the "sat" works better now. Changing brightness is another story, but I assume this is also work in progress.

**Comment by jannnfe** [comment]
> > "sat" basically works, but the colors (I just tested red) were only saturated when "bri" was between 16 and 39. Increasing "bri" from 39 to 40 then lead to a green tone with very low brightness. The color changes do really behave a bit odd.
> 
> Then you’re not running my code, or the device isn’t recognised so my code doesn’t kick in. Be sure to re-pair the light strip after upgrading to v2.7.0.

After upgrading to v2.7.0 and re-pairing the device i still have the situation described above. I deleted the fairy lights via Phoscon and added new. Do I have to delete directly via the deconz GUI?

**Comment by ebaauw** [comment]
What does the `/lights` resource for the Xmas light strip look like?  It should look like https://github.com/dresden-elektronik/deconz-rest-plugin/issues/3716#issuecomment-735467996

Deleting it from Phoscon and re-pairing should be enough, I think, but maybe you need to delete the record physically from the database.

**Comment by jannnfe** [comment]
The `/lights` resource looks like this:
```
"26": {
        "etag": "52a4c605a656533d9be0f8670d2f6ce0",
        "hascolor": true,
        "lastannounced": null,
        "lastseen": "2020-12-05T21:51Z",
        "manufacturername": "Heiman",
        "modelid": "TS0601",
        "name": "Wohnzimmer Weihnachtsbaum",
        "state": {
            "alert": "none",
            "bri": 254,
            "colormode": "xy",
            "effect": null,
            "hue": 4128,
            "on": false,
            "reachable": false,
            "sat": 16,
            "xy": [
                0.5499,
                0.393
            ]
        },
        "swversion": null,
        "type": "Color dimmable light",
        "uniqueid": "84:2e:14:ff:fe:05:dd:2c-01"
    }
```

Can you give me a little instruction please what i should do next? :)

**Comment by ebaauw** [comment]
The `manufacturername` has been mutilated to `"Heiman"`, causing my code not to recognise it as the Xmas lightstrip.  I don't quite know what's causing this, I think I read a report that it fixes itself after restarting deCONZ.  I don't know if this is the old database record being restored or a new record.  Do you know what was the previous resource id?  It should get a new id after deleting the resource and re-pairing the light strip.

**Comment by jannnfe** [comment]
I think I remember that the device was called Color dimmable light 25 at the beginning and after deleting and adding it was also called Color dimmable light 25. Maybe I have to try again?

**Comment by ebaauw** [comment]
I'm sorry your guess is as good as mine.  Your list above shows id 26 instead of 25, suggesting its a newly created resource, rather than a restored one.

**Comment by jannnfe** [comment]
I've just tried again using the REST API. The ID changes but is still recognized incorrectly. Shouldn't I have the right foundation with these specifications?
![Bildschirmfoto 2020-12-05 um 23 31 42](https://user-images.githubusercontent.com/30130742/101266755-505c3c00-3752-11eb-8213-d118a12ed146.png)

**Comment by ebaauw** [comment]
That's the right version, afaik.

**Comment by SwoopX** [comment]
Try reading the manufacturer name from basic cluster in deconz GUI. That should update the name in the DB as well as in Phoscon.

**Comment by Somberland** [comment]
The Thermostate-thing is also something that occurred (or better is still present) in phoscon. After adding the device to an group I can control it via phoscon. Will dump homebridge tomorrow.

![image](https://user-images.githubusercontent.com/5808345/101266888-9534a280-3753-11eb-8362-7084f97b156d.png)

![image](https://user-images.githubusercontent.com/5808345/101266896-ab426300-3753-11eb-852f-ee8ea4a01a8c.png)

**Comment by jannnfe** [comment]
> Try reading the manufacturer name from basic cluster in deconz GUI. That should update the name in the DB as well as in Phoscon.

That worked for me! Thanks everybody for your help :)

**Comment by SwoopX** [comment]
@Somberland The model identifier is also exposed by some thermostats, so that's the reason why.

**Comment by Smanar** [comment]
BTW I m interested by the log when including the device, if you still have "heiman" as manufacture name.

**Comment by jd-network** [comment]
Für die einzelnen Effekte können auch mehrere Farben definiert bzw. übermittelt werden:
{
    "effect": {
        "effect": "snake",
        "speed": 100,
        "colors": [
            {
                "r": 255,
                "g": 0,
                "b": 0
            },
            {
                "r": 0,
                "g": 255,
                "b": 0
            },
            {
                "r": 0,
                "g": 0,
                "b": 255
            }
        ]
    }
}


https://github.com/Koenkk/zigbee2mqtt.io/blob/master/docs/devices/HG06467.md

**Comment by r3pl1x** [comment]
> BTW I m interested by the log when including the device, if someone still have "heiman" as manufacture name.

![image](https://user-images.githubusercontent.com/67121863/101339377-a0d0b800-387e-11eb-89c3-b138a11f3925.png)

**Comment by Mimiix** [comment]
Hi there,

As this topic is slowly turning into a generic help topic, i am closing this off.  Phoscon needs to add support by itself, please open a issue in their repository.

@ebaauw  told me he wanted to have another PR in regards to this issue. He will keep this issue updated with the latest developments. 

If anyone has a issue with this device, feel free to open a user question issue or stop by in our discord server!

Thanks for anyone involved 👍 !

**Comment by ebaauw** [comment]
The PR is already there, https://github.com/dresden-elektronik/deconz-rest-plugin/issues/3716#issuecomment-736789205.  I just want it to be merged in the next (beta) release.

**Comment by ebaauw** [comment]
Latest PR doesn't work out as I'd hoped.  Setting the brightness when in colour mode might change the colour, see https://github.com/ebaauw/homebridge-hue/issues/860.  Need to revert supporting `bri_inc` and trying to keep _Current Level_ attribute in the _Level Control_ cluster reflecting the actual brightness.
